tests/fixtures/optimizer/tpc-ds/tpc-ds.sql

```diff
@@ -2029,18 +2029,7 @@ JOIN "date_dim" AS "date_dim"
   ON "date_dim"."d_year" = 2001
   AND "store_sales"."ss_sold_date_sk" = "date_dim"."d_date_sk"
 JOIN "household_demographics" AS "household_demographics"
-  ON "customer_demographics"."cd_demo_sk" = "store_sales"."ss_cdemo_sk"
-  AND "customer_demographics"."cd_education_status" = 'Advanced Degree'
-  AND "customer_demographics"."cd_education_status" = 'Primary'
-  AND "customer_demographics"."cd_education_status" = 'Secondary'
-  AND "customer_demographics"."cd_marital_status" = 'D'
-  AND "customer_demographics"."cd_marital_status" = 'M'
-  AND "customer_demographics"."cd_marital_status" = 'U'
-  AND "household_demographics"."hd_dep_count" = 1
-  AND "household_demographics"."hd_dep_count" = 3
-  AND "store_sales"."ss_hdemo_sk" = "household_demographics"."hd_demo_sk"
-  AND "store_sales"."ss_sales_price" <= 100.00
-  AND "store_sales"."ss_sales_price" >= 150.00
+  ON FALSE
```

Comment:
this looks wrong, what is this?

Reply:
Well I'm not sure if I'm missing some weird `NULL` shenanigans, but if you look at the `ON` clause it has the following conditions (prior to my changes):

```
  AND "customer_demographics"."cd_education_status" = 'Advanced Degree'
  AND "customer_demographics"."cd_education_status" = 'Primary'
```

These are contradicting, hence the `FALSE` value replacing the whole thing if I understand this correctly. Perhaps there was a bug prior to this PR?

Reply:
To be clear I'm still exploring `LEFT` and `RIGHT` joins in this PR, SQLite seems to avoid any transformations on the `ON` clause for them (see above discussion with Barak), so there's probably a catch.

Reply:
there was a previous bug and this exposes it, it should be fixed

but it's education_status = advanced degree or education_status = primary

Reply:
Fixed in https://github.com/tobymao/sqlglot/pull/2386/commits/d40fc163f593ff47a883428f9769a6e045776734